api/types/network/network.go

```diff
@@ -47,12 +47,21 @@ type PeerInfo struct {
 
 // EndpointSettings stores the network endpoint details
 type EndpointSettings struct {
-	// Configurations
+	// Configuration
+	NetworkID  string
 	IPAMConfig *EndpointIPAMConfig
 	Links      []string
 	Aliases    []string
+	DriverOpts map[string]string
+
 	// Operational data
-	NetworkID           string
+	EndpointOperationalData
```

Comment:
The real issue here is to mix both user-provided settings and state. The state should be derived from the settings, and not the other way around.

Reply:
Yeah, must 100% admit that I got lost on these types, and also confused whether `NetworkID` was a _config_ or _operational_ field. It was not reset in `cleanOperationalData` so assumed it was "config", but it was _listed_ under the "operational data" comment, so 🤷‍♂️